src/audio/sr_audio/sr_audio.c

```diff
@@ -0,0 +1,720 @@
+// SR_HIFI2_SetParameter(context, paramId, paramData, dataSize, dataSize, 1);
+//   or
+// SR_HIFI2_GetParameter(context, paramId, paramData, dataSize, dataSize, 1);
+//
+void SR_HIFI2_SetParameter(SR_HIFI2_Context_t* libCtx, int paramId, const uint8_t* paramData, int dataSize, int dataFragmentSize, int isFirstFragment);
+void SR_HIFI2_GetParameter(SR_HIFI2_Context_t* libCtx, int paramId, uint8_t* paramData, int dataSize, int dataFragmentSize, int isFirstFragment);
+
+// Toggles bypass mode (bypass = 1 for bypassing, 0 otherwise)
+void SR_HIFI2_Bypass(SR_HIFI2_Context_t* libCtx, int bypass);
+
+// Processing function.
+// frameCount: number of samples in each input/output buffers
+void SR_HIFI2_Process(SR_HIFI2_Context_t* libCtx, int frameCount, const int32_t* inputL, const int32_t* inputR, int32_t* outputL, int32_t* outputR);
+
+uint32_t xtensa_get_ccount();
+
+#ifdef _ADD_TEST_CLICK_AT48
+static void _add_test_click_at47(int32_t* samplesL, int32_t* samplesR) {
+  samplesL[47] = 0;
+  samplesR[47] = 0;
+}
+#endif
+
+#define maxChannelCount 2
+#define bufferFrameCount 48
+
+// SRAUDIO component private data
+struct comp_data {
+  struct comp_data_blob_handler* model_handler;
+  struct sof_sraudio_config* config;
+  size_t configSize;
+  enum sof_ipc_frame source_format; // source frame format
+  enum sof_ipc_frame sink_format;   // sink frame format
+  SR_HIFI2_Context_t sraudio_ctx;   // processing context
+  int32_t inProcessBuffer[maxChannelCount][bufferFrameCount];
+  int32_t outProcessBuffer[maxChannelCount][bufferFrameCount];
+  sraudio_func sraudio_func;        // processing function
+};
+
+#if CONFIG_FORMAT_S16LE
+static void sraudio_s16_default(const struct comp_dev* dev, const struct audio_stream* source, struct audio_stream* sink, uint32_t frameCount) {
+  struct comp_data* cd = comp_get_drvdata(dev);
+  int32_t ch, i, j;
+  uint32_t inChannelCount = source->channels;
+  uint32_t outChannelCount = sink->channels;
+  uint32_t inBufferSampleCount = bufferFrameCount * inChannelCount;
+  uint32_t outBufferSampleCount = bufferFrameCount * outChannelCount;
+  uint32_t tailFrameCount = frameCount;
+  uint32_t inOffset = 0;
+  uint32_t outOffset = 0;
+  SR_HIFI2_Context_t* ctx = &cd->sraudio_ctx;
+  for (; tailFrameCount >= bufferFrameCount; tailFrameCount -= bufferFrameCount, inOffset += inBufferSampleCount, outOffset += outBufferSampleCount) {
+    for (ch = inChannelCount; ch < maxChannelCount; ch++)
+      for (i = 0; i < bufferFrameCount; i++)
+        cd->inProcessBuffer[ch][i] = 0;
+    for (ch = 0; ch < inChannelCount; ch++) {
+      for (i = 0, j = ch + inOffset; i < bufferFrameCount; i++, j += inChannelCount)
+        cd->inProcessBuffer[ch][i] = (int32_t)*((int16_t*)audio_stream_read_frag_s16(source, j)) << 16;
+    }
+    SR_HIFI2_Process(ctx, bufferFrameCount, cd->inProcessBuffer[0], cd->inProcessBuffer[1], cd->outProcessBuffer[0], cd->outProcessBuffer[1]);
```

Comment:
What about Hifi3, 4 or 5 ?

Reply:
The prefix of this function is a historical rudiment from the early ports. The library itself is conditionally compiled for HiFi3/4/5.

Reply:
ok great :) Probably worth a comment so users will know they can get Hifi5 etc if needed.